### ISSUE TITLE
Windows machines use Python 3, but skydoc does not support it

### ISSUE THREAD

**Issue by filipesilva** [bazelbuild/continuous-integration#335]
Followup to https://github.com/bazelbuild/skydoc/issues/57#issuecomment-425130614

https://github.com/bazelbuild/skydoc does not support Python 3, but that is the Python used in Windows bazelbuild CI. It supports only Python 2.

Since Skydoc is a `bazelbuild/*` package, and also fairly common, it would be nice if the Windows CI could run skydoc targets instead of having an explicit exclusion.

**Comment by meteorcloudy** [comment]
Seems like passing `--python_path=<python2_binary>` will make it work.
I'll try to install python2 on Windows machines first, then we can try.

**Comment by filipesilva** [comment]
@meteorcloudy WDYT of having both python 2 and 3 installed, but having 2 as the default since it is what's required by skydoc? 

This way python 3 would be opt-in via `--python_path=` and Windows CI would not have to opt-in to python 2 to make skydoc.

**Comment by meteorcloudy** [comment]
I looked at our CI. Only tensorflow requires python3 on Windows, other projects should all work with python2.
@philwo Do you agree with we making python2 the default python on Windows?

**Comment by philwo** [comment]
Hi,

no, I’m sorry... Python 2 is basically EOL and I would not like to support it. Let’s instead fix Skydoc to work with Python 3.

Python 3 was released in *2008*. That was way enough time to migrate :)

**Comment by dslomov** [comment]
I would prefer doing no work on skydoc since stardoc is coming. Can we install python2, *not* make it default, and use it for skydoc? Migrating skydoc to python3 will be disruptive for its users, who will first need to scramble to update their python and then, few months later, scratch the whole thing entirely in favor of stardoc. /cc @c-parsons @laurentlb

**Comment by meteorcloudy** [comment]
Agree, I actually tried to make Skydoc Python3 compatible, but found it is not a trivia work.
> Can we install python2, not make it default, and use it for skydoc?

Yes, I'm going to do that.

**Comment by filipesilva** [comment]
Ah so we are already in a situation where python 2 is required for some things, and python 3 for others... That is unfortunate.

Is there a way to express that some rules should always use a certain python path/version besides the command line? Putting it on .bazelrc doesn't sound great since individual work machines won't have the same paths. If that's not available I guess the best way is to just append it to the windows targets.

So instead of 

```
platforms:
  windows:
    run_targets:
    - "@nodejs//:yarn"
    build_targets:
    # Escape hyphens on later args
    - "--"
    - "//..."
    # Cannot build the docsite on Windows, see
    # https://github.com/bazelbuild/skydoc/issues/58
    - "-//docs"
    test_targets:
    - "//..."
```

We'd have
```
platforms:
# ...
  windows:
    run_targets:
    - "@nodejs//:yarn"
    build_targets:
    # Escape hyphens on later args
    - "//..."
    - "--python_path=\"C:\python2\""
    test_targets:
    - "//..."
    - "--python_path=\"C:\python2\""
```


Going forward though, what's the preferred resolution for incompatible python versions? Is it to segregate build targets with different `--python_path`?

I don't feel that's a great experience myself. When skydoc didn't run on my machine I kept having to repeat commands because I forgot to add `-- -//docs/...` at the end of them. Some of those commands took a while to fail because skydoc still tried to build protobuff. Plus writing out the python path is more characters. If someone has to do the same because their workspace needs to use both python 2 and 3, that can get frustrating fast.

**Comment by dslomov** [comment]
@brandjon is working on better story for python2/3

**Comment by philwo** [comment]
I fixed Skydoc so that it works with Python 3: https://github.com/bazelbuild/skydoc/pull/110

**Comment by meteorcloudy** [comment]
Wow, this is amazing! Thanks, Philip!

**Comment by dslomov** [comment]
That works too :)

**Comment by filipesilva** [comment]
I think this is issue is closed by bazelbuild/skydoc#110. I updated the angular repos to use it (https://github.com/bazelbuild/rules_nodejs/pull/339 and https://github.com/bazelbuild/rules_typescript/pull/284) and the Windows CI seems to be quite green!
